compiler/rustc_trait_selection/src/traits/query/dropck_outlives.rs

```diff
@@ -196,11 +196,27 @@ where
                 debug!("dropck_outlives: ty from dtorck_types = {:?}", ty);
                 ty
             } else {
-                ocx.deeply_normalize(&cause, param_env, ty)?;
+                // Flush errors b/c `deeply_normalize` doesn't expect pending
+                // obligations, and we may have pending obligations from the
+                // branch above (from other types).
+                let errors = ocx.select_all_or_error();
+                if !errors.is_empty() {
+                    return Err(errors);
+                }
 
-                let errors = ocx.select_where_possible();
-                debug!("normalize errors: {ty} ~> {errors:#?}");
-                return Err(errors);
+                match ocx.deeply_normalize(&cause, param_env, ty) {
```

Comment:
pls add comment that we do this match to get `FulfillmentError`s to report in case `query_normalize` failed to avoid ICE due to a delayed bug